### PR TITLE
Add model-preparation-algorithm API & training tests

### PR DESCRIPTION
**Pull request by harimkang**
## What's in this PR
- Add model-preparation-algorithm (MPA) tests
- Update MPA tasks (inference task)
- Update submodule of model-preparation-algorithm

## Build for tests

- build for training tests
https://ci-ote.iotg.sclab.intel.com/job/ote/job/ote-test/1746/
[Validation-Report](http://validationreports.sclab.intel.com:8006/reports/build_number_report?test_session_build_number=ote-test-1746&environment=iotg-dev-workstation-12)

- build for API & precommit
https://ci-ote.iotg.sclab.intel.com/job/ote/job/ote-test/1745/
[Validation-Report](http://validationreports.sclab.intel.com:8006/reports/build_number_report?test_session_build_number=ote-test-1745&environment=iotg-dev-workstation-08)

During the test, there is a score drop issue during pot_evaluation (SKIPPED) 
in the class incremental learning for segmentation test and it is reported in the [CVS-84576](https://jira.devtools.intel.com/browse/CVS-84576)

### CODE REVIEW

**Comment by harimkang** [comment]
run ote-test

**Comment by chuneuny-emily** [comment]
run ote-test

File: external/model-preparation-algorithm/mpa_tasks/apis/classification/task.py [hunk 1]
```diff
@@ -229,17 +231,34 @@ def train(self,
               output_model: ModelEntity,
               train_parameters: Optional[TrainParameters] = None):
         logger.info('train()')
+        # Check for stop signal between pre-eval and training. 
+        # If training is cancelled at this point,
+        if self._should_stop:
```
Comment (chuneuny-emily) [inline external/model-preparation-algorithm/mpa_tasks/apis/classification/task.py:236]: What do you think of renaming from '_should_stop' to '_is_cancelled'?

**Review by chuneuny-emily — DISMISSED** [review]
Looks good for me.

File: external/model-preparation-algorithm/mpa_tasks/apis/classification/task.py (hunk 1 above)
Comment (harimkang) [reply external/model-preparation-algorithm/mpa_tasks/apis/classification/task.py:236]: The variable '_should_stop' is a common variable used in other tasks(mmdet/mmseg/deep-reid). So it doesn't seem appropriate to change.

Comment (chuneuny-emily) [reply external/model-preparation-algorithm/mpa_tasks/apis/classification/task.py:236]: I see. If the name is common one being used for the other backends, yes then it should be kept.

**Review by chuneuny-emily — DISMISSED** [review]
Looks good for me.

**Comment by harimkang** [comment]
During the real life test, there is a score drop issue during evaluation with FAILED status
in the class incremental learning for segmentation test and it is reported in the [CVS-84576](https://jira.devtools.intel.com/browse/CVS-84576)

**Comment by chuneuny-emily** [comment]
run ote-test

**Review by goodsong81 — COMMENTED** [review]
Looks good to me, but let's merge this PR after verifying MPA test cases are gathered well.

**Comment by harimkang** [comment]
run ote-test

**Review by chuneuny-emily — APPROVED** [review]
Looks good for me.

**Comment by harimkang** [comment]
-Update-
build for training tests PASSED
https://ci-ote.iotg.sclab.intel.com/job/ote/job/ote-test/1746/
[Validation-Report](http://validationreports.sclab.intel.com:8006/reports/build_number_report?test_session_build_number=ote-test-1746&environment=iotg-dev-workstation-12)

build for API & precommit PASSED
https://ci-ote.iotg.sclab.intel.com/job/ote/job/ote-test/1745/
[Validation-Report](http://validationreports.sclab.intel.com:8006/reports/build_number_report?test_session_build_number=ote-test-1745&environment=iotg-dev-workstation-08)

During the test, there is a score drop issue during pot_evaluation (now SKIPPED)
in the class incremental learning for segmentation test and it is reported in the [CVS-84576](https://jira.devtools.intel.com/browse/CVS-84576)

**Review by goodsong81 — APPROVED** [review]

**Pull request merged**